app/controllers/protocols_controller.rb

```diff
@@ -860,7 +860,8 @@ def load_organization_and_type
   def check_view_all_permissions
     load_organization_and_type
 
-    unless can_view_organization_protocols(@current_organization)
+    if @current_organization &&
+       !can_view_organization_protocols(@current_organization)
```

Comment:
If we check user's organizations in the view and disable protocols button on nav bar, then I think we don't need additional checks here and can keep original version which just throws 403 if user has no rights to view protocols:

unless can_view_organization_protocols(@current_organization)
  render_403
end